### PR TITLE
Add Targets field to TargetGroup resource

### PR DESCRIPTION
**Pull request by michaelhtm**
Issue [#2106](https://github.com/aws-controllers-k8s/community/issues/2106)

Description of changes:
* custom code to register and deregister targets
* add e2e tests

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by rushmash91 — APPROVED** [review]

**Review by a-hilaly — COMMENTED** [review]
Nice! Left a few comments below

File: templates/hooks/target_group/sdk_read_many_post_set_output.go.tpl [hunk 1]
```diff
@@ -0,0 +1,6 @@
+	err = rm.describeTargets(ctx, &resource{ko})
+	if err != nil {
+		return nil, err
+	}
+
+	rm.setStatusDefaults(ko)
```
[the remark this answers is not in the document]
Comment (michaelhtm) [reply templates/hooks/target_group/sdk_read_many_post_set_output.go.tpl:6]: yes..wasn't sure how to put the hook code above the setStatusDefaults

File: templates/hooks/target_group/sdk_update_post_set_output.go.tpl
[the remark this answers is not in the document]
Comment (michaelhtm) [reply templates/hooks/target_group/sdk_update_post_set_output.go.tpl:1]: YESS..doing this

**Review by a-hilaly — APPROVED** [review]
Thank you @michaelhtm !! 
/lgtm
/hold

**Pull request merged**